### PR TITLE
Add owner field to FeatureService

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**: This PR adds the `owner` field to `FeatureService`. It also cleans up `FeatureService` and `FeatureViewProjection`.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/feature_service.py [hunk 1]
```diff
@@ -19,21 +19,25 @@
...
+    A feature service defines a logical group of features from one or more feature views.
+    This group of features can be retrieved together during training or serving.
+
+    Attributes:
+        name: The unique name of the feature service.
+        feature_view_projections: A list containing feature views and feature view
+            projections, representing the features in the feature service.
+        description: A human-readable description.
+        tags: A dictionary of key-value pairs to store arbitrary metadata.
+        owner: The owner of the feature service.
```
Comment (woop) [inline sdk/python/feast/feature_service.py:31]: Should we give more details here? Should it be an email? Is this owner already registered elsewhere in the feature store? Can it be any string?

File: sdk/python/feast/feature_view_projection.py [hunk 2]
```diff
@@ -10,15 +10,26 @@
 
 @dataclass
 class FeatureViewProjection:
+    """
+    A feature view projection represents a set of features from a single feature view.
```
Comment (woop) [inline sdk/python/feast/feature_view_projection.py:14]: What about
```suggestion
    A feature view projection represents a selection of one more more features from a single feature view.
```

File: sdk/python/feast/feature_view_projection.py [hunk 3]
```diff
@@ -10,15 +10,26 @@
...
+
+    Attributes:
+        name: The unique name of the feature view projection.
```
Comment (woop) [inline sdk/python/feast/feature_view_projection.py:17]: Wait, but feature view projections aren't public facing objects right? So `name` is really the aliased name of the feature view, not the projection. Or am I missing something?

File: sdk/python/feast/feature_view_projection.py [hunk 4]
```diff
@@ -10,15 +10,26 @@
...
+        name_alias: An optional alias for the name.
+        features: The list of features represented by the feature view projection.
+        join_key_map: A map to modify join key columns during retrieval of this feature
+            view projection.
+    """
+
     name: str
-    name_alias: Optional[str]
-    features: List[Feature]
+    name_alias: str = ""
```
Comment (woop) [inline sdk/python/feast/feature_view_projection.py:25]: Is there a reason for this change?

File: sdk/python/feast/feature_view_projection.py [hunk 5]
```diff
@@ -10,15 +10,26 @@
...
+    features: List[Feature] = []
```
Comment (woop) [inline sdk/python/feast/feature_view_projection.py:26]: Doesn't this introduce a bug?

```class Service(object):
    data = []

s1 = Service()
s2 = Service()

s1.data.append(1)

print(s1.data)
[1]
print(s2.data)
[1]

s2.data.append(2)

print(s1.data)
[1, 2]
print(s2.data)
[1, 2]
```

File: sdk/python/feast/feature_service.py (hunk 1 above)
Comment (felixwang9817) [reply sdk/python/feast/feature_service.py:31]: Sure, added a note about email.

**Review by woop — APPROVED** [review]

**Pull request merged**